apps/frontend/app/(with-header)/profile/page.tsx

```diff
@@ -1,9 +1,14 @@
+import { headers } from 'next/headers';
+
 import { Profile } from 'src/components/auth/Profile';
 import { requireAuthorizedUser } from 'src/utils/auth.util';
 
 export default async function ProfilePage() {
+  const headersList = headers();
+  const pathname = headersList.get('x-pathname');
```

Comment:
You can call it inside `requireAuthorizedUser` function so there will be no need to duplicate this code on each page.